### PR TITLE
[single-article] Refactor class components to be functional component

### PR DESCRIPTION
**Pull request by vivianedias**
## What type of PR is this? (check all applicable)

- [x] Refactor
- [ ] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

As suggested in #17601 I'm separating the classes that were migrated to functional components into different PRs.

## Added/updated tests?

- [x] Yes
- [ ] No, and this is why: _please replace this line with details on why tests
      have not been included_
- [ ] I need help with writing tests

## [optional] What gif best describes this PR or how it makes you feel?

alt_text

### CODE REVIEW

File: app/javascript/modCenter/singleArticle/__tests__/singleArticle.test.jsx [hunk 1]
```diff
@@ -39,37 +39,24 @@ describe('<SingleArticle />', () => {
   });
 
   it('renders the article title', () => {
-    const { queryByText } = render(
+    const articleProps = getTestArticle();
+    const { getByRole } = render(
       <Fragment>
-        <SingleArticle {...getTestArticle()} toggleArticle={jest.fn()} />
+        <SingleArticle {...articleProps} toggleArticle={jest.fn()} />
```
Comment (Ridhwana) [inline app/javascript/modCenter/singleArticle/__tests__/singleArticle.test.jsx:45]: Nice one ✨

**Review by Ridhwana — APPROVED** [review]
Thanks @vivianedias ✨

File: app/javascript/modCenter/singleArticle/__tests__/singleArticle.test.jsx [hunk 2]
```diff
@@ -39,37 +39,24 @@ describe('<SingleArticle />', () => {
   });
 
   it('renders the article title', () => {
-    const { queryByText } = render(
+    const articleProps = getTestArticle();
```
Comment (aitchiss) [inline app/javascript/modCenter/singleArticle/__tests__/singleArticle.test.jsx:42]: (not blocking) - Looking at the specs in this file, I don't think there's any need for `getTestArticle` to be a function 🤔  If it wasn't, then we wouldn't need to call it every spec like this... but then it's not really related to this PR so could maybe be a later enhancement 🙂

File: app/javascript/modCenter/singleArticle/__tests__/singleArticle.test.jsx [hunk 3]
```diff
@@ -39,37 +39,24 @@ describe('<SingleArticle />', () => {
...
-        />
-      </Fragment>,
-    );
-    const text = getNodeText(
-      container.getElementsByClassName('article-title-heading')[0],
-    );
-    expect(text).toContain(getTestArticle().title);
+    expect(
+      getByRole('heading', { name: articleProps.title, level: 3 }),
+    ).toBeInTheDocument();
```
Comment (aitchiss) [inline app/javascript/modCenter/singleArticle/__tests__/singleArticle.test.jsx:55]: 😍

**Review by aitchiss — APPROVED** [review]
Love it! ✨

File: app/javascript/modCenter/singleArticle/__tests__/singleArticle.test.jsx (hunk 2 above)
Comment (vivianedias) [reply app/javascript/modCenter/singleArticle/__tests__/singleArticle.test.jsx:42]: Totally agree!

**Pull request merged**